Bio/Phylo/_utils.py

```diff
@@ -82,6 +82,107 @@ def build_subgraph(graph, top):
     return G
 
 
+def to_igraph(tree, vertex_attributes=None, edge_attributes=("color", "width")):
+    """Convert a Tree object to an igraph Graph.
+
+    The result is useful for graph-oriented analysis and interactive plotting
+    with matplotlib, Cairo, and plotly.
+
+    Requires python-igraph version 0.10.0 or later.
+
+    :Parameters:
+        vertex_attributes : sequence of strings
+            A sequence of strings containing the Clade properties that are to
+            be stored as vertex attributes, e.g. "name".
+        edge_attributes : sequence of strings
+            A sequence of strings containing the Clade properties that are to
+            be stored as edge attributes, e.g. "color" and "width".
+
+    In both cases, if a property is not found it will be ignored.
+    """
+    try:
+        import igraph as ig
+    except ImportError:
+        raise MissingPythonDependencyError(
+            "Install igraph if you want to use to_igraph."
+        ) from None
+
+    if ig.__version__ < "0.10":
+        raise MissingPythonDependencyError(
+            "Update igraph to 0.10 or later if you want to use to_igraph."
+        ) from None
+
+    # Count the leaves, thereby the total number of nodes in the tree
+    n_terminals = tree.count_terminals()
+    n_nodes = sum(1 for x in tree.find_clades())
+    n_edges = n_nodes - 1
+
+    # Empty tree
+    if n_terminals == 0:
+        return ig.Graph()
+
+    # NOTE: In igraph, adding all edges at once is much faster, so we prepare
+    # an edgelist and related attributes
+    def add_subtree(node, n_node, counter, edges, edge_attrs, vertex_attrs):
+        """Add edges from this subtree, breath-first."""
+        # NOTE: Bio.Phylo stores both vertex and edge attributes in the
+        # same place as "clade" attributes. However, the root node can have
+        # a vertex attribute but no edge attribute, hence the code blocks are
+        # in different places
+        for attrname in vertex_attrs:
+            if hasattr(node, attrname):
+                # Sometimes, only some tree nodes (e.g. leaves) have
+                # a certain attribute, e.g. a name
+                if len(vertex_attrs[attrname]) == 0:
+                    vertex_attrs[attrname] = [None for x in range(n_nodes)]
```

Comment:
`[None] * n_nodes` avoids the creation of a range and is probably more readable.

Reply:
done